_posts/2017-01-30-docker-frontend-tools.md

```diff
@@ -0,0 +1,146 @@
+date: 2017-01-30
+author: Dan Murphy
+tags: drupal drupal-planet docker pattern-lab front-end-dev drupal8
+summary: This post demonstrates how to build a Docker image that can be used in local Drupal development with the Pattern Lab Starter theme and/or with other common front-end applications such as npm, Gulp, and Bower.
+featured_image: "/blog/docker.png"
+featured_image_alt: "Docker placeholder image"
+featured_image_height: "983px"
+featured_image_width: "1474px"
+drupal_planet_summary: |
+  This post demonstrates how to build a Docker image that can be used in local Drupal development with the Pattern Lab Starter theme and/or with other common front-end applications such as npm, Gulp, and Bower.
+---
+
+## Overview
+
+We at Savas Labs have been using Docker for local development and CI environments for some time. Recently we wanted to test out [Phase 2's Pattern Lab Starter 8 theme](https://github.com/phase2/pattern-lab-starter). However, to use this theme locally we needed to build a new Docker image for running applications that the theme depends on. In this post, I'll share:
+
+- A `Dockerfile` used to build an image with Node, npm, PHP, and Composer installed
+- A `docker-compose.yml` configuration and Docker commands for running theme commands such as `npm start` from within the container
+
+Along the way, I'll also provide:
+
+- A quick overview of why we use Docker for local development
+- Tips for building custom images and running common front-end applications inside containers.
+
+## Background
+
+We at Savas switched to using Docker for local development last year and we love it - so much so that we even proposed [a Drupalcon session on our approach and experience](https://events.drupal.org/node/17250). Using Docker makes it incredibly easy for developers to quickly spin up consistent local development environments that match production. In the past we used Vagrant and virtual machines for these purposes, but we've found Docker to be much faster when switching between multiple projects.
+
+Usually we build our Docker images from scratch to perfectly match production environments. However, for agile development and rapid prototyping, we sometimes make use of public Docker images. In these cases we've relied on [Wodby's Docker4Drupal project](https://github.com/wodby/docker4drupal), which is "a set of docker containers optimized for Drupal."
+
+We plan to post a more in-depth series of posts covering Docker for local development and CI environments in the near future, so be on the look out for that!
+
+We're also fans of using style guides and atomic design to drive front-end development. So when we heard of the [Pattern Lab Starter Drupal theme](https://github.com/phase2/pattern-lab-starter) which leverages [Pattern Lab](http://patternlab.io/) (a tool for building pattern-driven user interfaces using atomic design), we were very excited to try it out.
```

Comment:
This is the same theme/URL referenced in the Overview section, but there you called it `Starter 8 Theme` and here it's `Starter Drupal Theme`. It confused me for a second and it might be good to make them match so it doesn't trip other people up too!